test/test_va_api_fixture.cpp

```diff
@@ -93,21 +95,61 @@ static VADisplay getDrmDisplay(int &fd)
 {
     typedef std::vector<std::string> DevicePaths;
     typedef DevicePaths::const_iterator DevicePathsIterator;
+    drmDevicePtr *devices;
+    int ret, max_devices;
 
-    static DevicePaths paths({"/dev/dri/renderD128", "/dev/dri/card0"});
-
-    const DevicePathsIterator endIt(paths.end());
-    for (DevicePathsIterator it(paths.begin()); it != endIt; ++it) {
-        fd = open(it->c_str(), O_RDWR);
-        if (fd < 0)
-            continue;
-        VADisplay disp = vaGetDisplayDRM(fd);
+    max_devices = drmGetDevices2(0, NULL, 0);
```

Comment:
The most pedantic solution is to call `drmGetDevices2` twice. The more pragmatic one is to pick a static array - say 32 or 64 entries. Should find some some time to improve the API + docs :sweat_smile:

Reply:
sure, done